betting-game-react/src/components/Navbar/Navbar.test.tsx

```diff
@@ -0,0 +1,148 @@
+
+import { createMemoryHistory } from 'history';
+import React from 'react';
+import { MemoryRouter, Router } from 'react-router-dom';
+
+import { fireEvent, render } from '@testing-library/react';
+
+import { ABOUTUS, LOGIN, MATCHES, STANDINGS } from '../../config/routes';
+import Navbar from './Navbar';
+import { NavbarButtonProps, navbarButtons } from './NavbarButtonProps';
+
+describe('<Navbar />', () => {
+  test('it should render a navbar with a logo, a login button and all navigation buttons', () => {
+    const history = createMemoryHistory();
+    history.push('/bad/route');
+    const { getByTestId } = render(
+      <MemoryRouter>
+        <Navbar />
+      </MemoryRouter>
+    );
+
+    const navbar = getByTestId('Navbar');
+    const navbarLogo = getByTestId('NavbarLogo');
+    let navbarButtonsArray: any[] = [];
+    navbarButtons.map((navbarButton: NavbarButtonProps, index: number) => {
+      navbarButtonsArray.push(
+        getByTestId(`${navbarButton.buttonTestIdPrefix}-NavButton`)
+      );
+    });
+
+    expect(navbar).toBeInTheDocument();
+    expect(navbarLogo).toBeInTheDocument();
+    navbarButtonsArray.forEach((navbarButton) => {
+      expect(navbarButton).toBeInTheDocument();
+    });
+  });
+
+  test('it should route when the "Matches" button was clicked', () => {
+    const { getByTestId } = render(<Navbar />, { wrapper: MemoryRouter });
+
+    const matchesButton = getByTestId(`Matches-NavButton`);
+    fireEvent.click(matchesButton);
+    const loadedComponent = getByTestId('Matches');
+
+    expect(loadedComponent).toBeInTheDocument();
+  });
+
+  test('it should route when the "Standings" button was clicked', () => {
+    const { getByTestId } = render(<Navbar />, { wrapper: MemoryRouter });
+
+    const matchesButton = getByTestId(`Standings-NavButton`);
+    fireEvent.click(matchesButton);
+    const loadedComponent = getByTestId('Standings');
+
+    expect(loadedComponent).toBeInTheDocument();
+  });
+
+  test('it should route when the "About us" button was clicked', () => {
+    const { getByTestId } = render(<Navbar />, { wrapper: MemoryRouter });
+
+    const matchesButton = getByTestId(`AboutUs-NavButton`);
+    fireEvent.click(matchesButton);
+    const loadedComponent = getByTestId('AboutUs');
+
+    expect(loadedComponent).toBeInTheDocument();
+  });
+
+  test('it should route when the "Login" button was clicked', () => {
+    const { getByTestId } = render(<Navbar />, { wrapper: MemoryRouter });
+
+    const matchesButton = getByTestId(`Login-NavButton`);
+    fireEvent.click(matchesButton);
+    const loadedComponent = getByTestId('Login');
+
+    expect(loadedComponent).toBeInTheDocument();
+  });
+
+  test('it should redirect to matches if the path does not exist', () => {
+    const history = createMemoryHistory();
+    history.push('/bad/route');
+    const { getByTestId } = render(
+      <Router history={history}>
+        <Navbar />
+      </Router>
+    );
```

Comment:
Good job splitting up the tests!

You are actually testing if the component is rendered - not if it actually redirects.
Maybe you could expect the current location to the route you are looking for?
`expect(location.pathname).toBe("/products");` 
as you can see [here](https://reactrouter.com/web/guides/testing)